local_server/local-server-generate-elevation-profile/src/main/java/com/esri/samples/local_server_generate_elevation_profile/LocalServerGenerateElevationProfileController.java

```diff
@@ -0,0 +1,577 @@
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.localserver.LocalGeoprocessingService;
+import com.esri.arcgisruntime.localserver.LocalGeoprocessingService.ServiceType;
+import com.esri.arcgisruntime.localserver.LocalServer;
+import com.esri.arcgisruntime.localserver.LocalServerStatus;
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.BasemapStyle;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Camera;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.LayerSceneProperties;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import com.esri.arcgisruntime.raster.HillshadeRenderer;
+import com.esri.arcgisruntime.raster.Raster;
+import com.esri.arcgisruntime.raster.RasterFunction;
+import com.esri.arcgisruntime.raster.RasterFunctionArguments;
+import com.esri.arcgisruntime.symbology.ColorUtil;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingFeatures;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingJob;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingString;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingTask;
+
+public class LocalServerGenerateElevationProfileController {
+
+  @FXML private Button drawPolylineButton;
+  @FXML private Button generateProfileButton;
+  @FXML private Button clearResultsButton;
+  @FXML private Label instructionsLabel;
+  @FXML private ProgressBar progressBar;
+  @FXML private SceneView sceneView;
+  @FXML private VBox vBox;
+
+  private ArcGISScene scene;
+  private FeatureCollection featureCollection;
+  private FeatureLayer featureLayer;
+  private GeoprocessingTask gpTask;
+  private GraphicsOverlay graphicsOverlay;
+  private LocalGeoprocessingService localGPService;
+  private Polyline polyline;
+  private Raster arranRaster;
+  private RasterLayer rasterLayer;
+  private Viewpoint rasterExtentViewPoint;
+
+  private static LocalServer server;
+
+  /**
+   * Called after FXML loads. Sets up scene.
+   */
+  public void initialize() {
+
+    try {
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a scene with a topographic basemap style
+      scene = new ArcGISScene(BasemapStyle.ARCGIS_HILLSHADE_DARK);
+
+      // create a new raster from local file and display it on the scene
+      arranRaster = new Raster(new File
```

Comment:
this is good to know: unfortunately it looks like it won't work here due to our need to retain "data.dir" for the sample viewer data paths. The Paths constructor looks to take that first argument as part of the path before adding on "samples-data" etc, but since it can't find that in the context of the sample, it reports the path as null.